vespa/datadog_checks/vespa/vespa.py

```diff
@@ -0,0 +1,125 @@
+                for metrics in service['metrics']:
+                    self._emit_metrics(service_name, metrics, instance_tags)
+
+            self.log.info("Forwarded {} metrics to hq for {} services".format(self.metric_count, self.services_up))
+            self.service_check(self.METRICS_SERVICE_CHECK, AgentCheck.OK, tags=instance_tags,
+                               message="Metrics collected successfully for consumer {}".format(consumer))
+        except Timeout as e:
+            self._report_metrics_error("Timed out connecting to Vespa's node metrics api: {}".format(e),
+                                       AgentCheck.CRITICAL, instance_tags)
+        except (HTTPError, InvalidURL, ConnectionError) as e:
+            self._report_metrics_error("Could not connect to Vespa's node metrics api: {}".format(e),
+                                       AgentCheck.CRITICAL, instance_tags)
+        except JSONDecodeError as e:
+            self._report_metrics_error("Error parsing JSON from Vespa's node metrics api: {}".format(e),
+                                       AgentCheck.CRITICAL, instance_tags)
+        except Exception as e:
+            self._report_metrics_error("Unexpected error: {}".format(e),
+                                       AgentCheck.WARNING, instance_tags)
+
+    def _report_metrics_error(self, msg, level, instance_tags):
+        self.log.warning(msg)
+        self.service_check(self.METRICS_SERVICE_CHECK, level, tags=instance_tags,
+                           message=msg)
+        self.service_check(self.PROCESS_SERVICE_CHECK, AgentCheck.WARNING, tags=instance_tags,
+                           message="Problem getting metrics from Vespa's node metrics api")
+        self.log.warning("Issued a warning to " + self.PROCESS_SERVICE_CHECK +
+                         " service check, as there is a problem getting metrics from Vespa.")
+
+    def _emit_metrics(self, service_name, metrics_elem, instance_tags):
+        """
+        Emit one metrics packet, which consists of a set of metrics that share the same set of dimensions.
+        :param metrics_elem: A (values, dimensions) tuple from the 'metrics' json array.
+        """
+        if 'values' not in metrics_elem:
+            return
+        metric_tags = self._get_tags(metrics_elem, service_name)
+        for name, value in metrics_elem['values'].items():
+            full_name = "vespa." + name
+            self._emit_metric(full_name, value, metric_tags + instance_tags)
+
+    def _emit_metric(self, name, value, tags):
+        self.log.debug("Emitting metric: {}, dimensions: {}".format(name, tags))
+        self.gauge(name, value, tags)
+        self.metric_count += 1
+
+    def _get_metrics_json(self, url, timeout):
+        """ Send rest request to metrics api and return the response as JSON
+        """
+        self.log.info("Sending request to {}".format(url))
+        response = requests.get(url, timeout=timeout)
```

Comment:
Hallo, I suggest using our request wrapper instead, it will give you the ability to use proxies, set addtional headers, disable ssl verification etc for free. You can see example PRs of how to include here: https://github.com/pulls?utf8=%E2%9C%93&q=is%3Apr+author%3AChristineTChen+request

Reply:
@hithwen Sorry, but I don't understand how the timeout is propagated into the wrapper. Is it enough to have a config option named 'timeout'? Or is it the local variable in the `check` function?

Does the 'get' call raise the same exceptions as `requests.get`?

Reply:
It is enough to have a config option called `timeout`. You can find the code for the requests wrapper here: https://github.com/DataDog/integrations-core/blob/master/datadog_checks_base/datadog_checks/base/utils/http.py#L40 all the options in standard fields become available for any integration using it. The wrapper ends up calling native `requests.get` in https://github.com/DataDog/integrations-core/blob/master/datadog_checks_base/datadog_checks/base/utils/http.py#L316